formatTest/typeCheckedTests/expected_output/attributes.re

```diff
@@ -405,9 +405,10 @@ type reactClass;
 
 /* "react-dom" shouldn't spread the attribute over multiple lines */
 [@bs.val] [@bs.module "react-dom"]
-external render: (reactElement, element) => unit;
+external render: (reactElement, element) => unit =
+  "render";
```

Comment:
this is regressing from #2464 in that the printer shouldn't print the `= "render";` part

Reply:
I'm not sure as the input for this particular output looks like this
```
/* "react-dom" shouldn't spread the attribute over multiple lines */
[@bs.val] [@bs.module "react-dom"]
external render : (reactElement, element) => unit = "render";
```

And the tests are passing, too. There's something else failing on the CI though which I don't understand.

Reply:
I think we agreed in the last PR that if the ident and the primitive name agree we can drop it on the floor, but I could be misremembering

Reply:
Or did I misunderstand the [assignment from Bob](https://github.com/facebook/reason/pull/2464#issuecomment-579232949)? 😄

Reply:
I would like to drop it if they are the same too btw. But I don't know all the in & outs of bs.

Reply:
Ah I didn’t read the discussion in the other issue.

I disagree with the outcome, but oh well.

Reply:
I think you all know this, but I still want to write this out for anyone following this discussion. 

> @anmonteiro I think we agreed in the last PR that if the ident and the primitive name agree we can drop it on the floor, but I could be misremembering

With this behavior, it would be a breaking change for BuckleScript.
The recommendation is to always explicitly write the string value, even if the identifier matches the value itself.

```
[@bs.val] [@bs.module "react-dom"]
external render : (reactElement, element) => unit = "render";
```

Previously, you could leave it out blank as well, but this will now yield a warning that this is discouraged. As @bobzhang  said, the reason is to prevent binding errors due to quick renaming of the identifier (with the punning, you wouldn't even know what's going on, especially if you are newcomer and never used the string representation for some reason).

Why does `[@bs.obj]` require a `= ""` then? This seems to be documented [here](https://bucklescript.github.io/docs/en/object-2#function):

```reason
[@bs.obj] external route: (
  ~_method:string,
  ~path:string,
  ~action:(list(string) => unit),
  ~options:Js.t({..})=?,
  unit
) => _ = "";
```

> Note: the = "" part at the end is just a dummy placeholder, due to syntactic limitations. It serves no purpose currently.

So i guess the whole goal of this general external syntax feature is to get rid of this confusing `[bs.obj]` syntax?